models/postModel.js

```diff
@@ -0,0 +1,41 @@
+const { doc, getDoc, setDoc, updateDoc, collection, query, orderBy, getDocs, where } = require('firebase/firestore')
+const db = require('../database');
+
+module.exports = {
+  addPost: async (postInfo) => {
+    const postRef = await doc(collection(db, 'posts'))
+    await setDoc(postRef, postInfo);
+  },
+  getPosts: async (category = null) => {
+    const postRef = await collection(db, 'posts')
+    let q = await query(postRef, orderBy('timestamp', 'desc'));
+    if (category) {
+      q = query(postRef, where('category', '==', category), orderBy('timestamp', 'desc'));
+    }
+    return await getDocs(q);
+  },
+  incrementLike: async (postid) => {
+    const postRef = await doc(db, 'posts', postid);
+    const postSnap = await getDoc(postRef);
+    const incrementedLike = postSnap.data().likes + 1;
+    if (incrementedLike >= 5) {
+      await updateDoc(postRef, { likes: 0, isApproved: true})
```

Comment:
Is it possible we get caught in a loop that resets likes to 0 every time we hit 5? Including after the post has been removed from the approval feed.